# Cancer Diagnosis Prediction 

## Feature Selection

Approach II: 
1. Train base boosting models - Light GBM, CatBoost, XGBoost. Plus one deep learning model - TabNet. 
2. Devide features by locations (`Region`, `Division`, `patient_state`) and apply Lasso Regression to each location separately to select important features
3. Apply a boosting model to each location. 

In [101]:
import pandas as pd

In [102]:
pd.set_option('display.max_columns', None)

In [103]:
train_base = pd.read_csv('train_1.csv')
test_base = pd.read_csv('test_1.csv')

In [104]:
train_base

patient_id  patient_zip3  patient_age   population       density  \
0          475714           924           84  31437.75000   1189.562500   
1          349367           928           62  39121.87879   2295.939394   
2          138632           760           43  21996.68333    626.236667   
3          617843           926           45  32795.32558   1896.220930   
4          817482           836           55  10886.26000    116.886000   
...           ...           ...          ...          ...           ...   
12897      674178           436           50  19413.05882   1196.805882   
12898      452909           945           50  30153.87952    976.289157   
12899      357486           926           61  32795.32558   1896.220930   
12900      935417           112           37  71374.13158  17326.407890   
12901      179597           336           29  30205.48148   1517.174074   

       age_median  age_under_10  age_10_to_19    age_20s    age_30s  \
0       30.642857     16.014286     15.542857  17.614286  14.014286   
1       38.200000     11.878788     13.354545  14.230303  13.418182   
2       37.906667     13.028333     14.463333  12.531667  13.545000   
3       42.871429     10.071429     12.135714  12.538095  12.464286   
4       43.473469     10.824000     13.976000   9.492000  10.364000   
...           ...           ...           ...        ...        ...   
12897   36.911765     12.876471     13.435294  14.394118  12.705882   
12898   42.135802     10.753086     12.714815  11.725926  13.101235   
12899   42.871429     10.071429     12.135714  12.538095  12.464286   
12900   36.476316     12.986842     11.318421  14.971053  17.255263   
12901   35.629630     11.696296     14.229630  16.592593  15.251852   

         age_40s    age_50s    age_60s   age_70s  age_over_80       male  \
0      11.614286  11.557143   7.571429  4.000000     2.100000  49.857143   
1      13.333333  14.060606  10.248485  5.951515     3.503030  49.893939   
2      12.860000  12.770000  11.426667  6.565000     2.811667  50.123333   
3      12.650000  14.847619  12.280952  8.216667     4.759524  49.066667   
4      12.600000  14.992000  14.836000  9.462000     3.466000  52.312000   
...          ...        ...        ...       ...          ...        ...   
12897  11.694118  13.329412  11.764706  6.188235     3.617647  48.264706   
12898  12.817284  13.301235  12.771605  8.413580     4.408642  49.727160   
12899  12.650000  14.847619  12.280952  8.216667     4.759524  49.066667   
12900  12.631579  11.460526   9.789474  6.000000     3.581579  47.668421   
12901  12.903704  11.629630   9.525926  5.466667     2.744444  49.696296   

          female    married   divorced  never_married   widowed  family_size  \
0      50.142857  36.571429  11.885714      47.114286  4.442857     3.928571   
1      50.106061  50.245455   9.827273      35.290909  4.651515     3.622727   
2      49.876667  55.753333  12.330000      27.195000  4.710000     3.260667   
3      50.933333  52.604762  11.623810      31.142857  4.623810     3.098095   
4      47.688000  57.882000  14.964000      21.760000  5.406000     3.352653   
...          ...        ...        ...            ...       ...          ...   
12897  51.735294  36.429412  15.700000      42.158824  5.705882     3.039412   
12898  50.272840  53.076543  10.912346      30.534568  5.466667     3.271125   
12899  50.933333  52.604762  11.623810      31.142857  4.623810     3.098095   
12900  52.331579  39.923684  10.239474      44.642105  5.186842     3.412105   
12901  50.303704  39.314815  14.922222      41.211111  4.544444     3.211154   

       family_dual_income  income_household_median  income_household_under_5  \
0               52.228571              52996.28571                  3.142857   
1               61.736364             102741.63640                  2.327273   
2               55.801667              85984.74138                  2.483333   
3               54.564286             120533.83330         

In [105]:
target = train_base['DiagPeriodL90D']
train_base.drop(columns=target.name, axis=1, inplace=True) 

In [109]:
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import Lasso
import numpy as np

def apply_lasso(df_slice, target, alpha=0.1,):
    # Splitting data into features (X) and target variable (y)
    y = target
    X = df_slice

    # Filter features based on correlation
    corr_matrix = X.corr().abs()
    upper_tri = corr_matrix.where(np.triu(np.ones(corr_matrix.shape), k=1).astype(bool))
    to_drop = [column for column in upper_tri.columns if any(upper_tri[column] > 0.8)]
    X_filtered = X.drop(columns=to_drop)

    # Standardize the Features
    scaler = StandardScaler()
    X_scaled = scaler.fit_transform(X)

    # Fit LASSO regression
    lasso_model = Lasso(alpha=alpha)
    lasso_model.fit(X_scaled, y)


    # Extract important features
    lasso_coefficients = lasso_model.coef_
    important_features = X.columns[np.abs(lasso_coefficients) > 0]

    return list(important_features)

In [112]:
from boosting_models import apply_catboost, apply_lightgbm, apply_xgboost
from tabnet import apply_tabnet
from tqdm import tqdm

In [113]:
important_features = apply_lasso(train_base, target, alpha=0.006)
important_features

['patient_zip3',
 'patient_age',
 'income_household_75_to_100',
 'home_value',
 'education_less_highschool',
 'self_employed',
 'commute_time',
 'health_uninsured',
 'patient_race_Black',
 'patient_race_White',
 'payer_type_COMMERCIAL',
 'payer_type_MEDICAID',
 'patient_state_AL',
 'patient_state_CO',
 'patient_state_LA',
 'patient_state_MN',
 'patient_state_NV',
 'patient_state_VA',
 'breast_cancer_diagnosis_code_1741',
 'breast_cancer_diagnosis_code_1742',
 'breast_cancer_diagnosis_code_1743',
 'breast_cancer_diagnosis_code_1744',
 'breast_cancer_diagnosis_code_1745',
 'breast_cancer_diagnosis_code_1746',
 'breast_cancer_diagnosis_code_1748',
 'breast_cancer_diagnosis_code_1749',
 'breast_cancer_diagnosis_code_1759',
 'breast_cancer_diagnosis_code_19881',
 'breast_cancer_diagnosis_code_C5001',
 'breast_cancer_diagnosis_code_C50119',
 'breast_cancer_diagnosis_code_C50421',
 'breast_cancer_diagnosis_code_C50811',
 'metastatic_cancer_diagnosis_code_C773',
 'metastatic_cancer_diagnosis_c

In [114]:
best_catboost_model, best_cat_params, best_cat_score = apply_catboost(train_base[important_features], target, cv=5)
print("Best Parameters for base model:", best_cat_params)
print("Best ROC for base model:", best_cat_score)

Best Parameters for base model: {'depth': 6, 'iterations': 200, 'learning_rate': 0.05}
Best ROC for base model: 0.8056441585217543


**Best previous result for CatBoost**

- Best Parameters for base model: {'depth': 4, 'iterations': 200, 'learning_rate': 0.05}
- Best ROC for base model: 0.8019676819045394

In [93]:
best_gbm_model, best_gbm_params, best_gmb_score = apply_lightgbm(train_base[important_features], target, cv=5)
print("Best Parameters for base model:", best_gbm_params)
print("Best ROC for base model:", best_gmb_score)

Best Parameters for base model: {'learning_rate': 0.05, 'n_estimators': 50, 'num_leaves': 31}
Best ROC for base model: 0.8002775778631868


**Best previous result for light GBM**

- Best Parameters for base model: {'learning_rate': 0.01, 'n_estimators': 150, 'num_leaves': 31}
- Best ROC for base model: 0.7959317174144424

In [15]:
best_xgb_model, best_xgb_params, best_xgb_score = apply_xgboost(train_base[important_features], target, cv=5)
print("Best Parameters for base model:", best_xgb_params)
print("Best ROC for base model:", best_xgb_score)

Best Parameters for base model: {'learning_rate': 0.05, 'max_depth': 4, 'n_estimators': 150}
Best ROC for base model: 0.8001561991412572


**Best previous result for XGBoost**
- Best Parameters for base model: {'learning_rate': 0.05, 'max_depth': 4, 'n_estimators': 150}
- Best ROC for base model: 0.8001561991412572

In [94]:
best_tabnet_model, best_tabnet_params, best_tabnet_score = apply_tabnet(train_base[important_features], target, cv=5)
print("Best Parameters for base model:", best_tabnet_params)
print("Best ROC for base model:", best_tabnet_score)

C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\pytorch_tabnet\abstract_model.py:82: UserWarning: Device used : cpu
  warnings.warn(f"Device used : {self.device}")


epoch 0  | loss: 0.75041 | val_0_auc: 0.7135  |  0:00:01s
epoch 1  | loss: 0.58128 | val_0_auc: 0.72773 |  0:00:02s
epoch 2  | loss: 0.53608 | val_0_auc: 0.74639 |  0:00:03s
epoch 3  | loss: 0.52254 | val_0_auc: 0.7538  |  0:00:04s
epoch 4  | loss: 0.51845 | val_0_auc: 0.75141 |  0:00:05s
epoch 5  | loss: 0.5143  | val_0_auc: 0.76124 |  0:00:06s
epoch 6  | loss: 0.51181 | val_0_auc: 0.75582 |  0:00:07s
epoch 7  | loss: 0.5028  | val_0_auc: 0.765   |  0:00:09s
epoch 8  | loss: 0.49913 | val_0_auc: 0.77345 |  0:00:10s
epoch 9  | loss: 0.49927 | val_0_auc: 0.77719 |  0:00:11s
epoch 10 | loss: 0.49722 | val_0_auc: 0.7791  |  0:00:12s
epoch 11 | loss: 0.4958  | val_0_auc: 0.78265 |  0:00:13s
epoch 12 | loss: 0.49228 | val_0_auc: 0.78815 |  0:00:14s
epoch 13 | loss: 0.49128 | val_0_auc: 0.7846  |  0:00:15s
epoch 14 | loss: 0.4884  | val_0_auc: 0.78728 |  0:00:17s
epoch 15 | loss: 0.48425 | val_0_auc: 0.79429 |  0:00:18s
epoch 16 | loss: 0.48244 | val_0_auc: 0.79749 |  0:00:19s
epoch 17 | los

C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\pytorch_tabnet\callbacks.py:172: UserWarning: Best weights from best epoch are automatically used!
  warnings.warn(wrn_msg)


Best Parameters for base model: None
Best ROC for base model: 0.8136765948486919


**Best previous result for TabNet**

- Best ROC for base model: 0.8136765948486919

In [115]:
def test_and_pack(test, important_features, best_model, name):
    scaler = StandardScaler()
    scaled = scaler.fit_transform(test[important_features])

    pred_y = best_tabnet_model.predict(scaled)

    preds = pd.DataFrame(pred_y, columns=['DiagPeriodL90D'])
    preds.insert(0, 'patient_id', test['patient_id'])

    preds.to_csv(f'{name}_result.csv', index=False)

In [117]:
# test_and_pack(test_base, important_features, best_tabnet_model, 'tabnet')

In [84]:
test_and_pack(test_base, important_features, best_catboost_model, 'cat')

## Train Models on Locations

In [ ]:
train = pd.read_csv('train_3.csv')
test = pd.read_csv('test_3.csv')

In [ ]:
def group_by_location(train, test, location_type):
    
    drop_columns = {
        'patient_state': ['patient_gender', 'Division', 'Region'],
        'Region': ['patient_gender', 'Division', 'patient_state'],
        'Division': ['patient_gender', 'patient_state', 'Region']
    }

    columns_to_drop = drop_columns.get(location_type, [])

    train = train.drop(columns_to_drop, axis=1)
    test = test.drop(columns_to_drop, axis=1)


    # Get unique states
    unique_locations = train[location_type].unique()

    # Create a dictionary to store DataFrames for each state
    locations_dfs = {}
    target_dfs = {}

    # Iterate over unique states
    for location in unique_locations:
        # Filter DataFrame for the current state
        location_df = train[train[location_type] == location].copy()
        location_df.drop([location_type, 'patient_id', 'DiagPeriodL90D'], axis=1, inplace=True)

        target_df = train[train[location_type] == location]['DiagPeriodL90D'].copy()
        
        # Store the filtered DataFrame in the dictionary with state as the key
        locations_dfs[location] = location_df
        target_dfs[location] = target_df

    return locations_dfs, target_dfs, train, test



division_dfs, division_target_dfs, division_train, division_test = group_by_location(train, test, 'Division')
region_dfs, region_target_dfs, regioon_train, region_test = group_by_location(train, test, 'Region')
state_dfs, state_target_dfs, state_train, state_test = group_by_location(train, test, 'patient_state')

In [12]:
def apply_models_by_location(location_type, locations_dfs, target_dfs, train, test):

    unique_locations = train[location_type].unique()

    # Example usage for each slice
    for location in tqdm(unique_locations):  # tqdm wrapper to show progress bar
        important_features = apply_lasso(locations_dfs[location], target_dfs[location], alpha=0.1)
        best_catboost_model, best_params, best_score = apply_catboost(locations_dfs[location][important_features], target_dfs[location], cv=2)
        print("Best Parameters for", location, ":", best_params)
        print("Best ROC for", location, ":", best_score)

In [13]:
apply_models_by_location('Division', division_dfs, division_target_dfs, division_train, division_test)

 12%|█▎        | 1/8 [00:05<00:38,  5.50s/it]

Best Parameters for Pacific : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for Pacific : 0.7084442052018601


 25%|██▌       | 2/8 [00:10<00:30,  5.11s/it]

Best Parameters for West South Central : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for West South Central : 0.6829556070277161


 38%|███▊      | 3/8 [00:15<00:24,  4.95s/it]

Best Parameters for Mountain : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for Mountain : 0.6994147617828097


 50%|█████     | 4/8 [00:20<00:20,  5.02s/it]

Best Parameters for Middle Atlantic : {'depth': 4, 'iterations': 100, 'learning_rate': 0.1}
Best ROC for Middle Atlantic : 0.7266128710787565


 62%|██████▎   | 5/8 [00:25<00:15,  5.19s/it]

Best Parameters for East North Central : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for East North Central : 0.7198174286474752


 75%|███████▌  | 6/8 [00:30<00:10,  5.12s/it]

Best Parameters for East South Central : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for East South Central : 0.754126776001776


 88%|████████▊ | 7/8 [00:35<00:04,  4.96s/it]

Best Parameters for South Atlantic : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for South Atlantic : 0.6569921755647057


100%|██████████| 8/8 [00:40<00:00,  5.03s/it]

Best Parameters for West North Central : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for West North Central : 0.7563179052767561


In [14]:
apply_models_by_location('Region', region_dfs, region_target_dfs, regioon_train, region_test)

 25%|██▌       | 1/4 [00:06<00:18,  6.28s/it]

Best Parameters for West : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for West : 0.7068186188330744


 50%|█████     | 2/4 [00:12<00:12,  6.17s/it]

Best Parameters for South : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for South : 0.6899812250037188


 75%|███████▌  | 3/4 [00:17<00:05,  5.77s/it]

Best Parameters for Northeast : {'depth': 4, 'iterations': 100, 'learning_rate': 0.1}
Best ROC for Northeast : 0.7266128710787565


100%|██████████| 4/4 [00:23<00:00,  6.00s/it]

Best Parameters for Midwest : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for Midwest : 0.7247134383437552


In [16]:
apply_models_by_location('patient_state', state_dfs, state_target_dfs, state_train, state_test)

  2%|▏         | 1/44 [00:04<03:20,  4.66s/it]

Best Parameters for CA : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for CA : 0.6812905606797031


  5%|▍         | 2/44 [00:09<03:12,  4.57s/it]

Best Parameters for TX : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for TX : 0.6711571358748778


  7%|▋         | 3/44 [00:14<03:16,  4.78s/it]

Best Parameters for ID : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for ID : 0.8350518368266848


  9%|▉         | 4/44 [00:19<03:12,  4.81s/it]

Best Parameters for NY : {'depth': 4, 'iterations': 200, 'learning_rate': 0.05}
Best ROC for NY : 0.7423914589968371


 11%|█▏        | 5/44 [00:23<03:03,  4.72s/it]

Best Parameters for IL : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for IL : 0.6962782147609669


 14%|█▎        | 6/44 [00:28<02:59,  4.73s/it]

Best Parameters for OR : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for OR : 0.7386853448275862


 16%|█▌        | 7/44 [00:33<03:00,  4.88s/it]

Best Parameters for MI : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for MI : 0.7598293679403989


 18%|█▊        | 8/44 [00:40<03:15,  5.44s/it]

Best Parameters for MT : {'depth': 4, 'iterations': 200, 'learning_rate': 0.05}
Best ROC for MT : 0.7300964187327823


 20%|██        | 9/44 [00:45<03:05,  5.30s/it]

Best Parameters for KY : {'depth': 4, 'iterations': 200, 'learning_rate': 0.05}
Best ROC for KY : 0.7834285187914517


 23%|██▎       | 10/44 [00:49<02:51,  5.04s/it]

Best Parameters for OH : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for OH : 0.7153386112106519


 25%|██▌       | 11/44 [00:53<02:37,  4.78s/it]

Best Parameters for IN : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for IN : 0.716645853552595


 27%|██▋       | 12/44 [00:58<02:28,  4.64s/it]

Best Parameters for FL : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for FL : 0.6610209362502941


 30%|██▉       | 13/44 [01:02<02:20,  4.55s/it]

Best Parameters for CO : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for CO : 0.5922437162784576


 32%|███▏      | 14/44 [01:07<02:23,  4.77s/it]

Best Parameters for VA : {'depth': 4, 'iterations': 100, 'learning_rate': 0.1}
Best ROC for VA : 0.6662415348196092


 34%|███▍      | 15/44 [01:13<02:26,  5.07s/it]

Best Parameters for GA : {'depth': 4, 'iterations': 200, 'learning_rate': 0.05}
Best ROC for GA : 0.7201004774175506


 36%|███▋      | 16/44 [01:18<02:24,  5.18s/it]

Best Parameters for KS : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for KS : 0.7613636363636364


 39%|███▊      | 17/44 [01:24<02:21,  5.22s/it]

Best Parameters for WI : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for WI : 0.7300780312124849


 41%|████      | 18/44 [01:28<02:09,  4.96s/it]

Best Parameters for PA : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for PA : 0.6609886930232444


 43%|████▎     | 19/44 [01:34<02:08,  5.14s/it]

Best Parameters for NM : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for NM : 0.7583333333333333


 45%|████▌     | 20/44 [01:39<02:02,  5.11s/it]

Best Parameters for WA : {'depth': 4, 'iterations': 100, 'learning_rate': 0.1}
Best ROC for WA : 0.692410043972544


 48%|████▊     | 21/44 [01:43<01:52,  4.88s/it]

Best Parameters for MN : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for MN : 0.7508097165991904


 50%|█████     | 22/44 [01:48<01:50,  5.03s/it]

Best Parameters for NC : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for NC : 0.7329940417080436


 52%|█████▏    | 23/44 [01:55<01:57,  5.61s/it]

Best Parameters for AL : {'depth': 6, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for AL : 0.7849570301493378


 55%|█████▍    | 24/44 [02:02<01:59,  5.98s/it]

Best Parameters for SC : {'depth': 4, 'iterations': 200, 'learning_rate': 0.05}
Best ROC for SC : 0.737780112044818


 57%|█████▋    | 25/44 [02:07<01:49,  5.75s/it]

Best Parameters for LA : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for LA : 0.7649456521739131


 59%|█████▉    | 26/44 [02:14<01:48,  6.02s/it]

Best Parameters for IA : {'depth': 6, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for IA : 0.7871984337501579


 61%|██████▏   | 27/44 [02:21<01:44,  6.16s/it]

Best Parameters for TN : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for TN : 0.70778430785356


 64%|██████▎   | 28/44 [02:27<01:40,  6.25s/it]

Best Parameters for MD : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for MD : 0.8801011029411765


 66%|██████▌   | 29/44 [02:34<01:36,  6.42s/it]

Best Parameters for HI : {'depth': 6, 'iterations': 200, 'learning_rate': 0.1}
Best ROC for HI : 0.8571428571428572


 68%|██████▊   | 30/44 [02:39<01:25,  6.13s/it]

Best Parameters for NE : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for NE : 0.75


 70%|███████   | 31/44 [02:44<01:13,  5.66s/it]

Best Parameters for AZ : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for AZ : 0.6540914883141707


 73%|███████▎  | 32/44 [02:49<01:06,  5.56s/it]

Best Parameters for MO : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for MO : 0.7932942708333334


 75%|███████▌  | 33/44 [02:56<01:05,  5.94s/it]

Best Parameters for NV : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for NV : 0.6923076923076923


 77%|███████▋  | 34/44 [03:02<00:59,  5.99s/it]

Best Parameters for SD : {'depth': 4, 'iterations': 200, 'learning_rate': 0.1}
Best ROC for SD : 0.6666666666666666


 80%|███████▉  | 35/44 [03:08<00:52,  5.84s/it]

Best Parameters for DC : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for DC : 0.75


 82%|████████▏ | 36/44 [03:13<00:46,  5.80s/it]

Best Parameters for AR : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for AR : 0.8243801652892562


 84%|████████▍ | 37/44 [03:21<00:43,  6.26s/it]

Best Parameters for UT : {'depth': 6, 'iterations': 300, 'learning_rate': 0.1}
Best ROC for UT : 0.9415584415584415


 86%|████████▋ | 38/44 [03:28<00:38,  6.46s/it]

Best Parameters for WV : {'depth': 6, 'iterations': 200, 'learning_rate': 0.05}
Best ROC for WV : 0.8263888888888888


C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\model_selection\_split.py:737: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=2.
  warnings.warn(
C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\model_selection\_validation.py:988: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\model_selection\_validation.py", line 977, in _score
    scores = scorer(estimator, X_test, y_test, **score_params)
  File "C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\metrics\_scorer.py", line 253, in __call__
    return self._score(partial(_cached_call, None), estimator, X, y_true, **_kwargs)
  File "C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\metrics\_scorer.py", line 350, in _score
    retu

Best Parameters for ND : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for ND : nan


 91%|█████████ | 40/44 [03:36<00:22,  5.51s/it]

Best Parameters for OK : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for OK : 0.736842105263158


C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\model_selection\_split.py:737: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=2.
  warnings.warn(
C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\model_selection\_validation.py:988: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\model_selection\_validation.py", line 977, in _score
    scores = scorer(estimator, X_test, y_test, **score_params)
  File "C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\metrics\_scorer.py", line 253, in __call__
    return self._score(partial(_cached_call, None), estimator, X, y_true, **_kwargs)
  File "C:\Users\Julia\AppData\Roaming\Python\Python310\site-packages\sklearn\metrics\_scorer.py", line 350, in _score
    retu

Best Parameters for WY : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for WY : nan


 95%|█████████▌| 42/44 [03:45<00:10,  5.13s/it]

Best Parameters for DE : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for DE : 0.64125


 98%|█████████▊| 43/44 [03:51<00:05,  5.39s/it]

Best Parameters for MS : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for MS : 0.8175913760152891


100%|██████████| 44/44 [03:57<00:00,  5.39s/it]

Best Parameters for AK : {'depth': 4, 'iterations': 100, 'learning_rate': 0.05}
Best ROC for AK : 1.0
